src/components/implementation/no_interface/cFE_booter/osfilesys.c

```diff
@@ -0,0 +1,308 @@
+#include "cFE_util.h"
+#include "osfilesys.h"
+
+struct FS fs[MAX_NUM_FS];
+struct fsobj files[MAX_NUM_FILES];
+
+/*
+ * Notes on this version:
+ * This is not the final implementation of a file system for the cFE on COS
+ * For now the file system is read only and uses static memory for fsobj and FS
+ * The next step is to write a dumb allocator/deallocator and use it for 
+ * new fs, fsobj, and file data
+ * I have been intentional to restrict memory specific details to this file
+ * osfilesys.h and osfiles.c 'should' not care about how FS, fsobj, and data
+ * are stored in memory.
+ */
+
+/******************************************************************************
+** Tar Level Methods
+******************************************************************************/
+
+/* 
+ * Loads the position in memory of linked
+ * Checks for badly linked or no linked tar file.
+ */
+uint32 tar_load()
+{
+    if(! &_binary_cFEfs_tar_size)                         return OS_FS_ERR_DRIVE_NOT_CREATED;
+    if(!_binary_cFEfs_tar_start)                          return OS_FS_ERR_DRIVE_NOT_CREATED;
+    if(&_binary_cFEfs_tar_end < &_binary_cFEfs_tar_start) return OS_FS_ERR_DRIVE_NOT_CREATED;
+    size    = (uint32) &_binary_cFEfs_tar_size;
```

Comment:
That would seem to be intuitive, however the names of the symbols which are produced by the linking process are misleading.  The first check is making the sure the size of the filesystem is greater than 0.  Second is making sure the actual symbols have been overwritten by linking process.  Third is sanity check that the end of the file should be after the beginning.  I agree that they are confusing but they correctly check what I am trying to check.  I am trying to rewrite them more clearly or at the least comment with their purposes